diff --git a/packages/privacy/src/privacy.cairo b/packages/privacy/src/privacy.cairo
--- a/packages/privacy/src/privacy.cairo
+++ b/packages/privacy/src/privacy.cairo
@@ -139,8 +139,9 @@ pub mod Privacy {
             assert_valid_execution_info(:execution_info);
             let server_actions = self.execute_view(:user_addr, :client_actions);
             // TODO: Test inner `is_valid_signature` panics.
-            assert_valid_signature(:user_addr, tx_info: execution_info.tx_info);
-            send_message_to_server(:server_actions);
+            //assert_valid_signature(:user_addr, tx_info: execution_info.tx_info);
+            //send_message_to_server(:server_actions);
+            panic(server_actions_to_panic_data(:server_actions));
         }

         fn execute_view(
